vllm/entrypoints/openai/serving_embedding.py

```diff
@@ -174,6 +176,7 @@ async def create_embedding(
         return response
 
     def _check_embedding_mode(self, embedding_mode: bool):
+        self._enabled = embedding_mode
```

Comment:
I think it would be cleaner to set `self._enabled` inside the initializer. Perhaps have this function return `embedding_mode` so we can assign `self._enabled = self._check_embedding_mode(...)`?

Reply:
Good idea!